### PR TITLE
Refactoring of the TV search service

### PR DESCRIPTION
**Pull request by 01sas**

### CODE REVIEW

File: apps/snapodds/src/services/auth/auth.service.ts [hunk 1]
```diff
@@ -23,8 +22,8 @@ export class AuthService {
     this.accessTokenUpdated.next(this.accessToken);
   }
 
-  refreshAccessToken(): Observable<string | undefined> {
-    if (this.hasValidAccessToken()) {
+  requireAccessToken(): Observable<string> {
+    if (this.accessToken != undefined && !this.isTokenExpired()) {
```
Comment (exaptis) [inline apps/snapodds/src/services/auth/auth.service.ts:26]: this line should be reverted to `if (this.hasValidAccessToken()) { ... }`

Comment (exaptis) [inline apps/snapodds/src/services/auth/auth.service.ts:19]: this line should be reverted to `if (this.hasValidAccessToken()) { ... }`

File: apps/snapodds/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -8,22 +8,22 @@ import { AuthService } from '../services/auth/auth.service';
 import { ApplicationConfigService } from '../services/config/application-config.service';
 import { LogLevel } from '../services/logger/log-level';
 import { NotificationService } from '../services/notification/notification.service';
-import { GoogleAnalyticsService } from '../services/tracking/google-analytics.service';
+import { AnalyticsService } from '../services/tracking/analytics.service';
```
Comment (exaptis) [inline apps/snapodds/src/app/app.component.spec.ts]: Renaming of the analytics service and it's methods should be either a separate commit or PR to make it easier to distinguish the changes.

File: apps/snapodds/src/app/odds/odds.component.ts [hunk 3]
```diff
@@ -37,23 +36,22 @@ export class OddsComponent implements OnChanges {
...
-      )
-      .subscribe({
-        next: (lineOdds) => {
-          this.lineOdds = lineOdds;
-          this.loading = false;
-        },
-        error: () => {
+    this.snapOddsFacade.getLineOdds(sportEventId).subscribe({
+      next: (lineOdds) => {
+        if (lineOdds.sportsBooks == undefined || lineOdds.sportsBooks.length === 0) {
```
Comment (exaptis) [inline apps/snapodds/src/app/odds/odds.component.ts:41]: I don't understand the benefits of this change. 
- Please use optional changing as it was before (you can read upon it here https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html). 
- By having this in the "success" event you've duplicated the lines 51:53 from the error block. If you don't like to throw an error at this point, then please move these 3 lines into a function and call it at #42 and #51.

**Review by exaptis — CHANGES_REQUESTED** [review]

File: apps/snapodds/src/app/app.component.spec.ts (hunk 2 above)
Comment (exaptis) [reply apps/snapodds/src/app/app.component.spec.ts]: Please perform the renaming in a separate branch as you are starting to refactor the code too, it makes it very difficult to spot the differences.

File: apps/snapodds/src/services/auth/auth.service.ts (hunk 1 above)
Comment (01sas) [reply apps/snapodds/src/services/auth/auth.service.ts:19]: No, it can not (at least I don't know how). I have changed the generic type of return value from Observable<string | undefined> to Observable<string> (because it should never happen that we will have an undefined to be pushed into this Observable, we only push a value when it exist and valid). But compiler in the case complains in the line this.accessTokenUpdated.next(this.accessToken) that it can not accept undefined value. Having a check like I have implemented removes this error.

File: apps/snapodds/src/app/app.component.spec.ts (hunk 2 above)
Comment (01sas) [reply apps/snapodds/src/app/app.component.spec.ts]: I have reverted renaming and will do it in a separate branch/PR.

File: apps/snapodds/src/app/odds/odds.component.ts (hunk 3 above)
Comment (01sas) [reply apps/snapodds/src/app/odds/odds.component.ts:41]: Optional chaining is not correct in this case, because actually from API we receive a response without sportsBooks collection (it is not an empty array, it is undefined) and in this case component does not show an empty message. I have added corresponding test case and introduced a function as you suggested.

File: apps/snapodds/src/services/auth/auth.service.ts (hunk 1 above)
Comment (exaptis) [reply apps/snapodds/src/services/auth/auth.service.ts:19]: The problem was that `this.accessToken != undefined` is a typeguard, so it knows in the branch that token has to be string. By calling the function this typeguard is lost, that's why I casted it to string.
